llvm/utils/llvm_push_pr.py

```diff
@@ -0,0 +1,653 @@
+        )
+        sys.exit(1)
+    printer.print("Prerequisites met.")
+
+
+def main():
+    """main entry point"""
+    parser = argparse.ArgumentParser(
+        description="Create and land a stack of Pull Requests."
+    )
+    GITHUB_REMOTE_NAME = "origin"
+    UPSTREAM_REMOTE_NAME = "upstream"
+    BASE_BRANCH = "main"
+
+    printer = Printer()
+    token = os.getenv("GITHUB_TOKEN")
+    default_prefix = "dev/"
+    if token:
+        # Create a temporary API client to get the user login
+        # We don't know the repo slug yet, so pass a dummy value.
+        temp_api = GitHubAPI("", printer, token)
+        try:
+            user_login = temp_api.get_user_login()
+            default_prefix = f"{user_login}/"
+        except requests.exceptions.RequestException as e:
+            printer.print(
+                f"Could not fetch user login from GitHub: {e}", file=sys.stderr
+            )
+
+    parser.add_argument(
+        "--base",
+        default=BASE_BRANCH,
+        help=f"Base branch to target (default: {BASE_BRANCH})",
+    )
+    parser.add_argument(
+        "--remote",
+        default=GITHUB_REMOTE_NAME,
+        help=f"Remote for your fork to push to (default: {GITHUB_REMOTE_NAME})",
+    )
+    parser.add_argument(
+        "--upstream-remote",
+        default=UPSTREAM_REMOTE_NAME,
+        help=f"Remote for the upstream repository (default: {UPSTREAM_REMOTE_NAME})",
+    )
+    parser.add_argument(
+        "--prefix",
+        default=default_prefix,
+        help=f"Prefix for temporary branches (default: {default_prefix})",
+    )
+    parser.add_argument(
+        "--draft", action="store_true", help="Create pull requests as drafts."
+    )
+    parser.add_argument(
+        "--no-merge", action="store_true", help="Create PRs but do not merge them."
+    )
+    parser.add_argument(
+        "--auto-merge",
+        action="store_true",
+        help="Enable auto-merge for each PR instead of attempting to merge immediately.",
+    )
+    parser.add_argument(
+        "--dry-run", action="store_true", help="Print commands without executing them."
+    )
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "-v", "--verbose", action="store_true", help="Print all commands being run."
+    )
+    group.add_argument(
+        "-q",
+        "--quiet",
+        action="store_true",
+        help="Print only essential output and errors.",
+    )
+
+    args = parser.parse_args()
+    if args.prefix and not args.prefix.endswith("/"):
+        args.prefix += "/"
+
+    printer = Printer(dry_run=args.dry_run, verbose=args.verbose, quiet=args.quiet)
+    check_prerequisites(printer)
+
+    # Get repo slug from git remote url
+    result = printer.run_command(
+        ["git", "remote", "get-url", args.remote],
+        capture_output=True,
+        text=True,
+        read_only=True,
+    )
+    url = result.stdout.strip()
+    match = re.search(r"github\.com[/:]([\w.-]+/[\w.-]+)", url)
```

Comment:
Why do we need to pull out the repo name? This should always just be the upstream monorepo.